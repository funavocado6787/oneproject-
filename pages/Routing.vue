<template>
  <v-container layout column>
    <v-layout row>
      <v-flex xs6>
        <h3>Nuxt Routing</h3>With Nuxt the routing table is automatically built automatically from the pages directory.
        Parameters are done via convetion of directory to be the route name and the parameter as the filename.
      </v-flex>
      <v-flex xs6 class="pl-3">
        <h4>Using Nuxt Routing</h4>
        <display-code :code="snippets.routing1" :oneLiner="true" />
      </v-flex>
    </v-layout>

    <v-layout row class="pt-3">
      <v-flex xs6>
        <h3>Vue JS - Example routing</h3>
        <p>The Basic routing table</p>
        <display-code :code="snippets.routing2" />
      </v-flex>
      <v-flex xs6 class="pl-3">
        <h4>Using the Routing</h4>

        <p>simply use the href or alternatively in the code use:</p>
            <display-code :code="snippets.routing3" :oneLiner="true" />
      </v-flex>
    </v-layout>
  </v-container>
</template>
<script>
import { mapGetters, mapActions } from 'vuex';

export default {
  name: "routing",
  computed: {
      ...mapGetters({ snippets: 'codeSnippet/getRoutingSnippets' }),
  },
  methods: {
      ...mapActions({ loadSnippets: 'codeSnippet/loadRoutingSnippets' })
  },
  head() {
    return {
      title: "Vue Router",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "GlennPacker.net vue router and routing"
        }
      ]
    };
  },
  created () {
      this.loadSnippets();
  },
};
</script>


