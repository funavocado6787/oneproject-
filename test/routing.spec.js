import { RouterLinkStub, shallowMount, createLocalVue, mount } from '@vue/test-utils';
import Routing from '../pages/Routing.vue';
import Vue from 'vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

Vue.use(Vuetify)
import DisplayCode from '../components/displayCode.vue'
Vue.component(DisplayCode.name, DisplayCode);

const localVue = createLocalVue()
localVue.use(Vuex)

describe('routing.vue', () => {
    let store, state;

    beforeEach(() => {
        state = {
            routingSnippets: {}
        }
        store = new Vuex.Store({
            modules: {
                codeSnippet: {
                    namespaced: true,
                    actions: {
                        loadRoutingSnippets({commit}) {
                            commit('setRoutingSnippets', {
                                routing1: 'routing1',
                                routing2: 'routing2',
                                routing3: 'routing3'
                             });
                        }
                    },
                    getters: {
                        getRoutingSnippets: state => state.routingSnippets
                    },
                    state,
                    mutations: {
                        setRoutingSnippets(state, data) {
                            state.routingSnippets = data;
                        }
                    }
                }
            }
        })
    })

    test('Smoke Test - check home page renders by looking for some hard coded text', async () => {
        const wrapper = shallowMount(Routing, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Nuxt Routing');
    })

    test('routing snippet load call was made on creation', async () => {
        const wrapper = shallowMount(Routing, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Nuxt Routing');
        expect(Object.keys(state.routingSnippets).length).toBe(3);
    })

    test('routing made use of snippet 1', async () => {
        const wrapper = mount(Routing, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('routing1');
    })

    test('routing made use of snippet 2', async () => {
        const wrapper = mount(Routing, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('routing2');
    })

    test('routing made use of snippet 3', async () => {
        const wrapper = mount(Routing, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('routing3');
    })

    test('match snapshot', async () => {
        const wrapper = mount(Routing, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.element).toMatchSnapshot()
    })
})
