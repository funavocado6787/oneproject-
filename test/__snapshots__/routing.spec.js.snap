// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routing.vue match snapshot 1`] = `
<div
  class="container layout column"
>
  <div
    class="layout row"
  >
    <div
      class="flex xs6"
    >
      <h3>
        Nuxt Routing
      </h3>
      With Nuxt the routing table is automatically built automatically from the pages directory.
      Parameters are done via convetion of directory to be the route name and the parameter as the filename.
    
    </div>
     
    <div
      class="flex pl-3 xs6"
    >
      <h4>
        Using Nuxt Routing
      </h4>
       
      <div
        id="app"
      >
        <div
          class="prism-editor-wrapper my-editor one-liner"
        >
          <div
            class="prism-editor__container"
          >
            <textarea
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="prism-editor__textarea"
              data-gramm="false"
              data-testid="textarea"
              placeholder=""
              spellcheck="false"
            />
            <pre
              class="prism-editor__editor"
              data-testid="preview"
            >
              routing1
              <br />
            </pre>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="layout pt-3 row"
  >
    <div
      class="flex xs6"
    >
      <h3>
        Vue JS - Example routing
      </h3>
       
      <p>
        The Basic routing table
      </p>
       
      <div
        id="app"
      >
        <div
          class="prism-editor-wrapper my-editor"
        >
          <div
            class="prism-editor__container"
          >
            <textarea
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="prism-editor__textarea"
              data-gramm="false"
              data-testid="textarea"
              placeholder=""
              spellcheck="false"
            />
            <pre
              class="prism-editor__editor"
              data-testid="preview"
            >
              routing2
              <br />
            </pre>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="flex pl-3 xs6"
    >
      <h4>
        Using the Routing
      </h4>
       
      <p>
        simply use the href or alternatively in the code use:
      </p>
       
      <div
        id="app"
      >
        <div
          class="prism-editor-wrapper my-editor one-liner"
        >
          <div
            class="prism-editor__container"
          >
            <textarea
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="prism-editor__textarea"
              data-gramm="false"
              data-testid="textarea"
              placeholder=""
              spellcheck="false"
            />
            <pre
              class="prism-editor__editor"
              data-testid="preview"
            >
              routing3
              <br />
            </pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
